### PR TITLE
Use JDK source if path provided

### PR DESCRIPTION
**Pull request by dharrigan**
If Java JDK sources are already installed (normally via some package
management), then instead of downloading the sources, allow the user to
point to where the sources can be found on the filesystem.

-=david=-

- [X] I created a issue to discuss the problem I am trying to solve or there is already a open issue.

### CODE REVIEW

File: lib/src/clojure_lsp/feature/java_interop.clj [hunk 1]
```diff
@@ -120,16 +120,24 @@
...
+       (spit dest-jdk-result-file "1")))
+    (.getCanonicalPath dest-jdk-file) db))
+
+(defn retrieve-and-analyze! [{:keys [db]}]
+  (let [jdk-source-uri (settings/get db [:java :jdk-source-uri] default-jdk-source-uri)
+        download-jdk-source? (settings/get db [:java :download-jdk-source?] false)]
+    (when-let [jdk-source-uri' (or (and (shared/plain-uri? jdk-source-uri)
+                                        (shared/filename->uri jdk-source-uri db))
+                                   (when download-jdk-source? (download-jdk-source jdk-source-uri db)))]
+      (logger/info (str "[JDK] Analyzing JDK source found at " jdk-source-uri "..."))
```
Comment (ericdallo) [inline lib/src/clojure_lsp/feature/java_interop.clj:141]:
```suggestion
      (logger/info (str "[JDK] Analyzing JDK source found at " jdk-source-uri' "..."))
```

File: lib/src/clojure_lsp/feature/java_interop.clj [hunk 2]
```diff
@@ -120,16 +120,24 @@
...
+      (analyze-jdk-source! jdk-source-uri' db)
+      (logger/info (str "[JDK] Source found at " jdk-source-uri " analyzed successfully.")))))
```
Comment (ericdallo) [inline lib/src/clojure_lsp/feature/java_interop.clj:143]: We already mention `jdk-source-uri` on line 141, I think we don't need to mention again?

Comment (dharrigan) [reply lib/src/clojure_lsp/feature/java_interop.clj:143]: np, will remove.

**Review by ericdallo — APPROVED** [review]
Thank you!

**Comment by snoe** [comment]
Isn't source.zip placed in JAVA_HOME typically? Seems like that should be looked at first.

**Comment by ericdallo** [comment]
@snoe yep, I'm adding support for finding automatically right now :)

**Comment by dharrigan** [comment]
@snoe bear in mind that some distro's don't set java_home, for example on Arch it's not set.

**Pull request merged**